analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicValue.cs

```diff
@@ -35,8 +35,22 @@ public sealed record SymbolicValue
         public static readonly SymbolicValue True = NotNull.WithConstraint(BoolConstraint.True);
         public static readonly SymbolicValue False = NotNull.WithConstraint(BoolConstraint.False);
 
+        private readonly ImmutableDictionary<Type, SymbolicConstraint> constraints = ImmutableDictionary<Type, SymbolicConstraint>.Empty;
+        private int? constraintsHashCode;
```

Comment:
It is and will be hash code for the entire class. No matter how much mess we add.
```suggestion
        private int? hashCode;
```